Assets/Scripts/EnemySpawner.cs

```diff
@@ -32,7 +48,27 @@ public class EnemySpawner : MonoBehaviour {
 		Object prefab = whatToSpawn[index];
 		if (prefab) {
 			GameObject enemy = Instantiate (prefab, new Vector3 (x, y, 0), new Quaternion (0, 0, 0, 0)) as GameObject;
-			enemy.GetComponent<FollowTarget> ().target = this.gameObject;
+			enemy.GetComponent<FollowTarget> ().target = target;
 		}
 	}
+
+	/// <summary>
+	/// Attachs a profile to the current enemy spawner.
+	/// </summary>
+	/// <param name="toAttach">To attach.</param>
+	public void attachProfile(WaveProfile toAttach){
```

Comment:
And about the time between the waves, definitely an interesting approach, but its out of the scope of this task according to the description